commcare-cloud/commcare_cloud.py

```diff
@@ -0,0 +1,125 @@
+# coding=utf-8
+from __future__ import print_function
+import getpass
+import os
+from six.moves import input
+from argparse import ArgumentParser
+import subprocess
+from clint.textui import puts, colored
+
+
+def ask(message):
+    return 'y' == input('{} [y/N]'.format(message))
+
+
+class AnsiblePlaybook(object):
+    @staticmethod
+    def make_parser(parser):
+        parser.add_argument('--skip-check', action='store_true', default=False)
+        parser.add_argument('--branch', default='master')
+        parser.add_argument('playbook')
+
+    @staticmethod
+    def run(args, unknown_args):
+        check_branch(args)
+
+        def ansible_playbook(environment, playbook, vault_password, *cmd_args):
+            cmd = (
+                'ANSIBLE_CONFIG={}'.format(os.path.expanduser('~/.commcare-cloud/ansible/ansible.cfg')),
+                'ansible-playbook',
+                os.path.expanduser('~/.commcare-cloud/ansible/{playbook}'.format(playbook=playbook)),
+                '-u', 'ansible',
+                '-i', os.path.expanduser('~/.commcare-cloud/inventory/{env}'.format(env=environment)),
+                '-e', '"@{}"'.format(os.path.expanduser('~/.commcare-cloud/vars/{env}/{env}_vault.yml'.format(env=environment))),
+                '-e', '"@{}'.format(os.path.expanduser('~/.commcare-cloud/vars/{env}/{env}_public.yml"'.format(env=environment))),
+                '--vault-password-file=/bin/cat',
+                '--diff',
+            ) + cmd_args
+            print(' '.join(cmd))
+            p = subprocess.Popen(' '.join(cmd), stdin=subprocess.PIPE, shell=True)
+            p.communicate(input='{}\n'.format(vault_password))
+            return p.returncode
```

Comment:
Any reason not to run the playbook using the python API?

Reply:
Yeah, nothing solid, I was just interested in mapping this directly onto the commands we're used to running, and if we use the API I'm less familiar with the behavior, and depending on the level we hook into we wouldn't get the command-line checks on args etc. I felt like I had a lot more certainty about how it would behave if I accessed it through the shell.